covalent_slurm_plugin/slurm.py

```diff
@@ -433,64 +309,179 @@ def _query_result(self, result_filename: str, task_results_dir: str) -> Any:
                 "-e",
                 remote_result_filename,
             ],
-            capture_output=True,
         )
-        if proc.returncode != 0:
-            raise FileNotFoundError(proc.returncode, proc.stderr, remote_result_filename)
+        if return_code != 0:
+            raise FileNotFoundError(return_code, proc_stderr, remote_result_filename)
 
         # Copy result file from backend to Covalent server
-        subprocess.run(
+        await self.run_async_subprocess(
             [
                 "rsync",
                 "-e",
                 f"ssh -i {self.ssh_key_file} -o StrictHostKeyChecking=no "
                 "-o UserKnownHostsFile=/dev/null -o LogLevel=ERROR",
                 f"{self.username}@{self.address}:{remote_result_filename}",
-                task_results_dir + "/",
+                f"{task_results_dir}/",
             ],
-            check=True,
-            capture_output=True,
         )
 
         # Copy stdout, stderr from backend to Covalent server
-        subprocess.run(
+        await self.run_async_subprocess(
             [
                 "rsync",
                 "-e",
                 f"ssh -i {self.ssh_key_file} -o StrictHostKeyChecking=no "
                 "-o UserKnownHostsFile=/dev/null -o LogLevel=ERROR",
                 f"{self.username}@{self.address}:{self.options['output']}",
-                task_results_dir + "/",
+                f"{task_results_dir}/",
             ],
-            check=True,
-            capture_output=True,
         )
-        subprocess.run(
+
+        await self.run_async_subprocess(
             [
                 "rsync",
                 "-e",
                 f"ssh -i {self.ssh_key_file} -o StrictHostKeyChecking=no "
                 "-o UserKnownHostsFile=/dev/null -o LogLevel=ERROR",
                 f"{self.username}@{self.address}:{self.options['error']}",
-                task_results_dir + "/",
+                f"{task_results_dir}/",
             ],
-            check=True,
-            capture_output=True,
         )
 
         local_result_filename = os.path.join(task_results_dir, result_filename)
-        with open(local_result_filename, "rb") as f:
-            result, exception = pickle.load(f)
-        os.remove(local_result_filename)
+        async with aiofiles.open(local_result_filename, "rb") as f:
+            contents = await f.read()
+            result, exception = pickle.loads(contents)
```

Comment:
Same comment as on the SSH plugin PR -- let's replace this with `json.loads` in a followup issue.